src/metabase/lib/drill_thru/common.cljc

```diff
@@ -11,6 +13,20 @@
       :lib/type
       (= :mbql.stage/mbql)))
 
+(defn native-stage?
+  "Is this query stage a native stage?"
+  [query stage-number]
+  (-> (lib.util/query-stage query stage-number)
+      :lib/type
+      (= :mbql.stage/native)))
+
+(defn prepare-query
```

Comment:
FIXME! Better semantics/different namespace?

Reply:
I've moved these to `lib.query` and refactored them somewhat.